res/schema.xml

```diff
@@ -504,4 +504,15 @@ METADATA
       ALTER TABLE library ADD COLUMN coverart_digest BLOB;
     </sql>
   </revision>
+  <revision version="34" min_compatbile="1">
+    <description>Add Macros</description>
+    <sql>
+      CREATE TABLE macros (
+        id INTEGER PRIMARY KEY AUTOINCREMENT,
+        track_id INTEGER NOT NULL REFERENCES library(id),
+        label TEXT DEFAULT '' NOT NULL,
+        state INTEGER DEFAULT 0 NOT NULL,
```

Comment:
Default values are only a workaround when altering tables, not for new tables.

Reply:
Why?

Reply:
Think about it! Hint: What is the meaning of 0?

Reply:
That the Macro is disabled and not looped, i.e. nothing enabled

Reply:
This was a retorical question.

Reply:
Why do we need a default value anyway? We should set it explicitly in the INSERT INTO statement. We do need default values when we alter a table because we try to be backwards compatible, so when you switch back to an older version the old INSERT INTO statement won't know about this field.

Reply:
@Holzhaus Correct. It is neither needed nor desirable.

Reply:
Thanks, I wasn't clear on this :)